pkg/storage/chunk/fetcher/fetcher.go

```diff
@@ -173,6 +183,11 @@ func (c *Fetcher) FetchChunks(ctx context.Context, chunks []chunk.Chunk, keys []
 	if err != nil {
 		level.Warn(log).Log("msg", "error fetching from cache", "err", err)
 	}
+
+	for _, buf := range cacheBufs {
+		chunkFetchedSize.With(labels.FromStrings("source", "cache").Map()).Observe(float64(len(buf)))
```

Comment:
`WithValues` should be more performant than creating a new labels set and turning it into a map.

Reply:
hah, you beat me

Reply:
Doh! Thanks for spotting this